bin/hdfcoinc/pycbc_coinc_statmap

```diff
@@ -19,112 +22,216 @@ class fw(object):
+                         '0 for no hierarchical removal. Choose 1 to do at '
+                         'most 1 hierarchical removal, and so on. The code '
+                         'will write the triggers out for whichever comes '
+                         'first: that there are no more hierarchical removals '
+                         'to do, or the max allowed amount of hierarchical '
+                         'removals has been reached. [default=0]')
+parser.add_argument('--store-last-h-rm-background', action='store_true',
+                    default='store_false',
+                    help='Toggle whether the user wants to store the '
+                         'inclusive background from the last iteration '
+                         'of hierarchical removal. If not indicated '
+                         'the final inclusive background triggers will '
+                         'be the inclusive background prior to hierarchical '
+                         'removal. (Note: The output file still stores '
+                         'the inclusive background from each hierarchical '
+                         'removal, but they are not stored as -final- results.'
+                         ' [default=false]')
 parser.add_argument('--output-file')
 args = parser.parse_args()
 pycbc.init_logging(args.verbose)
 
+# Require that the user does not input negative number of iterations to
+# hierarchically remove foreground triggers. This is required for a later
+# hack in the while loop if the user wants to do continuous hierarchical
+# removal.
+if args.max_hier_removal < 0:
+   logging.warn('ERROR: Cannot hierarchically remove triggers %s '
+                'times.' % args.max_hier_removal)
+   logging.warn('Exiting.')
+   sys.exit()
+
+# Set an index to keep track of how many hiearrchical removals we want to do.
+# This number will be used to keep track of which inclusive background we are
+# writing to file
+h_iterations = 0
+
 logging.info("Loading coinc triggers")    
-d = pycbc.io.StatmapData(files=args.coinc_files)   
-logging.info("We have %s triggers" % len(d.stat))
-fore_locs = d.timeslide_id == 0
-ft1, ft2 = d.time1[fore_locs], d.time2[fore_locs]
-vt = (ft1 + ft2) / 2.0
-veto_start, veto_end = vt - args.veto_window, vt + args.veto_window
-veto_time = abs(veto.start_end_to_segments(veto_start, veto_end).coalesce())  
+all_trigs = pycbc.io.StatmapData(files=args.coinc_files)
+
+logging.info("We have %s triggers" % len(all_trigs.stat))
+fore_locs = all_trigs.timeslide_id == 0
+
+# array of foreground trigger times for ifo 
+fore_time1 = all_trigs.time1[fore_locs]
+fore_time2 = all_trigs.time2[fore_locs]
+
+# array of average times of triggers from ifo1 and ifo2
+ave_fore_time = (fore_time1 + fore_time2) / 2.0
+
+# veto start and end time around every average foreground trigger time to
+# window around.
+veto_start_time = ave_fore_time - args.veto_window
+veto_end_time = ave_fore_time + args.veto_window
```

Comment:
I suggest a find and replace on ``veto`` to ``remove_window`` as these aren't really vetoes.